command/agent/acl_endpoint_test.go

```diff
@@ -221,6 +221,49 @@ func TestHTTP_ACLTokenBootstrap(t *testing.T) {
 	})
 }
 
+func TestHTTP_ACLTokenBootstrapOperator(t *testing.T) {
+	ci.Parallel(t)
+	conf := func(c *Config) {
+		c.ACL.Enabled = true
+		c.ACL.PolicyTTL = 0 // Special flag to disable auto-bootstrap
+	}
+	httpTest(t, conf, func(s *TestAgent) {
+		// Provide token
+		args := structs.ACLTokenBootstrapRequest{
+			BootstrapSecret: "2b778dd9-f5f1-6f29-b4b4-9a5fa948757a",
+		}
+
+		buf := encodeReq(args)
+
+		// Make the HTTP request
+		req, err := http.NewRequest("PUT", "/v1/acl/bootstrap", buf)
+		if err != nil {
+			t.Fatalf("err: %v", err)
+		}
+
+		// Since we're not actually writing this HTTP request, we have
+		// to manually set ContentLength
+		req.ContentLength = -1
```

Comment:
Wow this took me a while to figure out: since we're never actually writing this HTTP request, the stdlib code that sets ContentLength is never set!

Therefore in tests we need to set ContentLength to some nonzero value since tests do not use a real HTTP server.

Reply:
I am glad it is finally solved it kept me up a few nights :-)